### PR TITLE
Screengrab class

### PR DESCRIPTION
**Pull request by houghj16**
## Summary:
- This pull request adds a check for the screen recorder cache path before starting the recording. If the cache path is empty or invalid, an exception will be thrown and the recording will not start. This is to prevent the recording from failing silently or corrupting the video file.

### Files changed:
- ScreenRecorder.cs: Added a method ValidateCachePath that checks if the cache path is empty or invalid and throws an exception if so. Called this method in the StartRecording method before creating the video file.
- ScreenRecorderTest.cs: Added a unit test TestValidateCachePath that verifies the behavior of the ValidateCachePath method.

### Changes:
- Added a new exception class ScreenRecorderException that inherits from Exception and has a custom message property.
- Added a new method ValidateCachePath in the ScreenRecorder class that checks if the cache path is empty or invalid and throws a ScreenRecorderException if so.
- Added a call to ValidateCachePath in the StartRecording method before creating the video file.
- Added a new unit test TestValidateCachePath in the ScreenRecorderTest class that verifies the behavior of the ValidateCachePath method.

### Impact:
- This change will improve the user experience by preventing the recording from failing silently or corrupting the video file if the cache path is empty or invalid.
- This change will also make the code more robust and maintainable by handling the edge case of an empty or invalid cache path.

### Test plan:
- [ ] Run the unit test TestValidateCachePath and verify that it passes.
- [ ] Run the screen recorder with a valid cache path and verify that it works as expected.
- [ ] Run the screen recorder with an empty or invalid cache path and verify that it throws an exception and does not start the recording.

### CODE REVIEW

File: ShareX.ScreenCaptureLib/ScreenRecording/ScreenGrab.cs [hunk 1]
```diff
@@ -33,9 +33,10 @@
 
 namespace ShareX.ScreenCaptureLib
 {
-    public class ScreenRecorder : IDisposable
+    // The screengrab class is used to record the screen and save it to a file
+    public class ScreenGrab : IDisposable
     {
-        public bool IsRecording { get; private set; }
+        public bool On { get; private set; }
```
Comment (houghj16) [inline ShareX.ScreenCaptureLib/ScreenRecording/ScreenGrab.cs:42]: New comment

File: ShareX/ScreenRecordManager.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public static class ScreenRecordManager
     {
         public static bool IsRecording { get; private set; }
 
-        private static ScreenRecorder screenRecorder;
+        private static ScreenGrab screenGrab; // screenGrab is used to record screen
```
Comment (houghj16) [inline ShareX/ScreenRecordManager.cs:42]: add documentation here
